com.github.fries1234.ncsa-mosaic.json

```
@@ -0,0 +1,112 @@
+{
+    "app-id": "com.github.fries1234.ncsa-mosaic",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "21.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "Mosaic",
+    "finish-args": [
+        "--filesystem=home",
+        "--share=network",
+        "--socket=x11"
+    ],
+    "modules": [
+        {
+            "name": "xbitmaps",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/data/xbitmaps-1.1.2.tar.bz2",
+                    "sha256": "b9f0c71563125937776c8f1f25174ae9685314cbd130fb4c2efce811981e07ee"
+                }
+            ]
+        },
+        {
+            "name": "libxmu",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libXmu-1.1.3.tar.bz2",
+                    "sha256": "9c343225e7c3dc0904f2122b562278da5fed639b1b5e880d25111561bac5b731"
+                }
+            ]
+        },
+        {
+            "name": "libxft",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libXft-2.3.4.tar.bz2",
+                    "sha256": "57dedaab20914002146bdae0cb0c769ba3f75214c4c91bd2613d6ef79fc9abdd"
+                }
+            ]
+        },
+        {
+            "name": "openmotif",
+            "buildsystem": "autotools",
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://downloads.sourceforge.net/project/motif/Motif%202.3.8%20Source%20Code/motif-2.3.8.tar.gz",
+                    "sha256": "859b723666eeac7df018209d66045c9853b50b4218cecadb794e2359619ebce7"
+
+                }
+            ],
+            "post-install": [
+                "rm -rf /app/lib/*.a"
+            ]
+        },
+        {
+            "name": "mosaic",
+            "buildsystem": "simple",
+            "build-commands": [
+                "CFLAGS=-fcommon meson build",
+                "ninja -C build",
+                "install -Dm755 build/src/Mosaic -t /app/bin",
```

```suggestion
            "buildsystem": "meson",
            "build-options": {
                "cflags": "-fcommon",
            },
            "post-install": [
                "install -Dm755 build/src/Mosaic -t /app/bin",
```

Since you use meson now.
Note that the post-install will be done form the build dir, so path for source files might need adjustments.